Hardware/Hardware/CardReader/CardReaderSoftwareUpdateState.swift

```diff
@@ -0,0 +1,17 @@
+/// A type that represents the possible states for software update
+public enum CardReaderSoftwareUpdateState {
+    /// An optional update is available
+    case available
+
+    /// A mandatory update has started
+    case started(cancelable: FallibleCancelable?)
+
+    /// The update is being installed
+    case installing(progress: Float)
+
+    /// The update has finished installing
+    case completed
+
+    /// No update available
+    case none
```

Comment:
Would a separate state for `unknown` be appropriate - i.e. we haven't yet checked if an update is available? Separate issue is fine for that too.

I bring it up because below we initialize with `none` and I would have expected something more like `unknown`

Reply:
I'm not sure `unknown` is useful in this case, or that its usefulness outweighs the added complexity. Given the frequency of updates, it seems OK to assume that `none` is a good default. After all, if we haven't checked yet if there is an update, there is no update available. Maybe there is one in the server, but the same can be true if we just checked 10 seconds ago and Stripe just pushed an update now. 

More generally, I think an unknown state is useful if it's going to be reflected in the UI somehow, which I don't think it's the case here.

Reply:
A check-in-progress for a (optional) update is in the current UI, but we can cross this bridge in a later PR at which time it might continue to be useful to bind to .unknown to show an activity indicator while we checking